### ISSUE TITLE
Type Error Creating NamedTuple from Unpacked Tuple

### ISSUE THREAD

**Issue by emj1991** [microsoft/pyright#737]
**Describe the bug**
I have a tuple for which Pyright understands the types of each value. When I unpack the tuple using the * operator to create a namedtuple that has types, Pyright reports and error.

**To Reproduce**
Create new project with code below.

**Expected behavior**
No error from Pyright

**Screenshots or Code**
```
from typing import NamedTuple, List, Tuple


X = NamedTuple('X', [('a', int), ('b', str)])

q: List[Tuple[int, str]] = [(1, 'a'), (2, 'b'), (3, 'c')]

[X(*item) for item in q]
```

Hovering over item shows that the types are understood to be Tuple[int, str]. If you make this `X(item[0], item[1])` then the error goes away.

**VS Code extension or command-line**
Running VSCode version 1.46.0
Pyright plugin version 1.1.43
**Additional context**

**Comment by erictraut** [comment]
Thanks for the suggestion. This was moderately difficult to implement given the complexity of argument/parameter matching in Python, but I think I got it working without introducing regressions. This will be in the next published version of Pyright.

**Comment by erictraut** [comment]
This is now fixed in version 1.1.44, which I just published.
